x-pack/plugins/maps/public/connected_components/map/mb/get_initial_view.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { INITIAL_LOCATION } from '../../../../common/constants';
+import { Goto, MapCenterAndZoom } from '../../../../common/descriptor_types';
+import { MapSettings } from '../../../reducers/map';
+
+export async function getInitialView(
+  goto: Goto | null,
+  settings: MapSettings
+): Promise<MapCenterAndZoom | null> {
+  if (settings.initialLocation === INITIAL_LOCATION.LAST_SAVED_LOCATION) {
+    return goto && goto.center ? goto.center : null;
+  } else if (settings.initialLocation === INITIAL_LOCATION.FIXED_LOCATION) {
+    return {
+      lat: settings.initialLat,
+      lon: settings.initialLon,
+      zoom: settings.initialZoom,
+    };
+  } else if (settings.initialLocation === INITIAL_LOCATION.BROWSER_LOCATION) {
+    return await new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(
+        // success callback
+        pos => {
+          resolve({
+            lat: pos.coords.latitude,
+            lon: pos.coords.longitude,
+            zoom: settings.initialZoom,
+          });
+        },
+        // error callback
+        () => {
+          // eslint-disable-next-line no-console
+          console.warn('Unable to fetch browser location for initial map location');
+          resolve(null);
+        }
+      );
+    });
+  } else {
```

Comment:
Should we default to last_saved_location instead (?). If this doesn't get merged before 7.8, MapSettings might be sparsely filled in, and it is the current default behavior as well.

Reply:
Map settings will always have proper default values even if the settings are from a previous kibana instance. The redux reducer merges the existing settings with default settings.

https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/reducers/map.js#L189
```
case SET_MAP_SETTINGS:
      return {
        ...state,
        settings: { ...getDefaultMapSettings(), ...action.settings },
      };
```